teloxide_tests/src/mock_bot.rs

```diff
@@ -70,63 +76,60 @@ fn find_chat_id(value: Value) -> Option<i64> {
+#[derive(Default)]
+pub struct State {
+    pub files: Vec<File>,
+    pub responses: Responses,
+    pub messages: Messages,
+}
+
+impl State {
+    pub fn reset(&mut self) {
+        self.responses = Responses::default();
     }
-    if let MessageKind::Common(ref mut message_kind) = message.kind {
-        if let Some(ref mut reply_message) = message_kind.reply_to_message {
-            add_message(reply_message);
+
+    fn add_message(&mut self, message: &mut Message) {
+        let max_id = self.messages.max_message_id();
+        if message.id.0 <= max_id || self.messages.get_message(message.id.0).is_some() {
+            message.id = MessageId(max_id + 1);
+        }
+        if let Some(file_meta) = find_file(serde_json::to_value(&message).unwrap()) {
+            let file = File {
+                meta: file_meta,
+                path: "some_path.txt".to_string(), // This doesn't really matter
+            };
+            self.files.push(file);
+        }
+        if let MessageKind::Common(ref mut message_kind) = message.kind {
+            if let Some(ref mut reply_message) = message_kind.reply_to_message {
+                self.add_message(reply_message);
+            }
         }
+        self.messages.add_message(message.clone());
     }
-    MESSAGES.add_message(message.clone());
-}
-
-async fn stop_server() {
-    let client = reqwest::Client::new();
-    let _ = client
-        .post(format!(
-            "http://127.0.0.1:{}/stop/false",
-            MockBot::PORT.lock().unwrap().clone()
-        ))
-        .send()
-        .await;
 }
 
 /// A mocked bot that sends requests to the fake server
 /// Please check the `new` function docs and [github examples](https://github.com/LasterAlex/teloxide_tests/tree/master/examples) for more information.
-#[allow(dead_code)]
 pub struct MockBot {
     /// The bot with a fake server url
     pub bot: Bot,
     /// The thing that dptree::entry() returns
     pub handler_tree: UpdateHandler<Box<dyn std::error::Error + Send + Sync + 'static>>,
-    // Mutexes is here to not worry about mut references, its easier for the user without them
     /// Updates to send as user
-    pub updates: Mutex<Vec<Update>>,
+    pub updates: Vec<Update>,
     /// Bot parameters are here
-    pub me: Mutex<Me>,
+    pub me: Me,
     /// If you have something like a state, you should add the storage here using .dependencies()
-    pub dependencies: Mutex<DependencyMap>,
-    /// Caught responses from the server
-    pub responses: Mutex<Option<Responses>>,
-    /// Stack size used for dispatching
-    pub stack_size: Mutex<usize>,
-    bot_lock: Mutex<Option<MutexGuard<'static, ()>>>, // Maybe in the future ill make something like an atomic
-                                                      // bool that says if the bot is locked or not, and implement a custom Drop trait
+    pub dependencies: DependencyMap,
+
+    current_update_id: AtomicI32,
+    stack_size: usize,
```

Comment:
`.stack_size(&mut self, stack_size: usize)` is needed, or at least the field needs to be public

Reply:
Oops, yeah that should've been public